### PR TITLE
Use DbPrefix in index prefix

### PR DESCRIPTION
**Pull request by rnewson**

### CODE REVIEW

**Comment by rnewson** [comment]
Note that this isn't quite right, I think I'm intruding on parts of the primary key space.

File: src/search3_util.erl [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
--export([design_doc_to_index/2, get_doc/2]).
+-export([design_doc_to_index/3, get_doc/2]).
 
 -include_lib("couch/include/couch_db.hrl").
 
 -include("search3.hrl").
 
-design_doc_to_index(#doc{id=Id,body={Fields}}, IndexName) ->
+design_doc_to_index(#{db_prefix := DbPrefix}, #doc{id=Id,body={Fields}}, IndexName) ->
```
Comment (tonysun83) [inline src/search3_util.erl:10]: ahhh I see, the main change here is that we're using `DbPrefix` which is already encoded, whereas previously I was just using the `DbName`

**Comment by rnewson** [comment]
ignore previous, it's right in the sense that the search indexes are wiped out when the db is deleted, this caused a seeming bug in search3-java but it turned out was exactly the right behaviour (the open index was deleted and so the next update failed as the write.lock was no longer valid).

**Review by davisp — APPROVED** [review]
+1

**Review by davisp — APPROVED** [review]
+1

**Pull request merged**